src/lib/snark_params/scan_state_constants.ml

```diff
@@ -0,0 +1,46 @@
+[%%import
+"../../config.mlh"]
+
+(* in the config files, we either have:
+
+    - transaction_capacity_log_2 and work_delay exp explicitly, or
+    - those are calculated from the tps_goal and num_snark_workers
+
+   use the config boolean scan_state_calc_params to distinguish those cases
+
+ *)
+
+module type S = Scan_state_constants_intf.S
+
+[%%if
+scan_state_calc_params]
+
+open Core_kernel
+
+[%%inject
+"tps_goal", scan_state_tps_goal]
+
+[%%inject
+"num_snark_workers", scan_state_num_snark_workers]
+
+[%%inject
+"block_window_duration", block_window_duration]
+
+let max_coinbases = 2
+
+let max_user_commands_per_block = tps_goal * block_window_duration
```

Comment:
`block_window_duration` is in milliseconds. So you want to divide by 1000 here.

Reply:
`F#` statically verifies units, just sayin'.